cassiopeia/src/main/java/org/apache/tika/parser/ocr/TesseractOCRParser.java

```diff
@@ -83,271 +86,290 @@
-            waitThread.interrupt();
-            process.destroy();
-            Thread.currentThread().interrupt();
-            throw new TikaException("TesseractOCRParser interrupted", e);
-
-        } catch (ExecutionException e) {
-            // should not be thrown
-            messageHandler.handleMessage("TesseractOCRParser attempting to retrive result of aborted task.", e, MessageType.ERROR);
-        } catch (TimeoutException e) {
-            waitThread.interrupt();
-            process.destroy();
-            throw new TikaException("TesseractOCRParser timeout", e);
-        }
-
-    }
-
-    /**
-     * Reads the contents of the given stream and write it to the given XHTML
-     * content handler. The stream is closed once fully processed.
-     *
-     * @param stream
-     *          Stream where is the result of ocr
-     * @param xhtml
-     *          XHTML content handler
-     * @throws SAXException
-     *           if the XHTML SAX events could not be handled
-     * @throws IOException
-     *           if an input error occurred
-     */
-    private void extractOutput(InputStream stream, XHTMLContentHandler xhtml) throws SAXException, IOException {
-
-        xhtml.startDocument();
-        xhtml.startElement("div");
-        try (Reader reader = new InputStreamReader(stream, UTF_8)) {
-            char[] buffer = new char[1024];
-            for (int n = reader.read(buffer); n != -1; n = reader.read(buffer)) {
-                if (n > 0)
-                    xhtml.characters(buffer, 0, n);
-            }
-        }
-        xhtml.endElement("div");
-        xhtml.endDocument();
-    }
-
-    /**
-     * Starts a thread that reads the contents of the standard output or error
-     * stream of the given process to not block the process. The stream is closed
-     * once fully processed.
-     */
-    private void logStream(final String logType, final InputStream stream, final File file) {
-        new Thread() {
-            public void run() {
-                Reader reader = new InputStreamReader(stream, UTF_8);
-                StringBuilder out = new StringBuilder();
-                char[] buffer = new char[1024];
-                try {
-                    for (int n = reader.read(buffer); n != -1; n = reader.read(buffer))
-                        out.append(buffer, 0, n);
-                } catch (IOException e) {
-                    messageHandler.handleMessage("Could not read input stream.", e, MessageType.ERROR);
-                } finally {
-                    IOUtils.closeQuietly(stream);
-                }
-
-                String msg = out.toString();
-                LogFactory.getLog(TesseractOCRParser.class).debug(msg);
-            }
-        }.start();
-    }
-
-    static String getTesseractProg() {
-        return System.getProperty("os.name").startsWith("Windows") ? "tesseract.exe" : "tesseract";
-    }
+	@Autowired
+	private ISystemMessageHandler messageHandler;
```

Comment:
this won't work. since this class is not managed by spring, autowiring doesn't do anything. (I know it's wasn't you who put that there, but you're the lucky one to fix it ;)